### PR TITLE
[MBL-983] Already Reported Project Label

### PR DESCRIPTION
**Pull request by scottkicks**
# 📲 What

Replace the "Report this project to Kickstarter" cell with a SwiftUI view that dynamically shows either the original label or a label informing the user that they've already reported the project.

# 🤔 Why

If a user has already reported a project there's no need to do it again.

# 🛠 How

Instead of adding a label to the `ProjectPamphletSubpage`, we can build the new labels with SwiftUI and add it as its own section to the project page's table view datasource. Giving us more control and more SwiftUI.

Uses the new `flagging` property on Projects to conditionally render the labels. 
The new "Already reported" label should not be tappable and include 2 hyperlinks.

Followed [Android's implementation](https://kickstarter.slack.com/files/U0558SFCTL5/F05RP9YLE5C/screenshot_20230911-141236.png) as a design reference. 

# 👀 See

| iPad | iPhone |
| --- | --- |
| Simulator Screen Recording - iPad Pro (12 9-inch) (6th generation) - 2023-09-25 at 11 23 26 | Simulator Screen Recording - iPhone 14 - 2023-09-25 at 11 27 19 |

# ✅ Acceptance criteria
You'll need to ask someone on web team A to add your staging account email to the project "Undercover:: The cover art of underglow comics - 200+ pages". This is a test project that has been marked as flagged, but you'll still need it associated with your account so that you see the "Already reported" label. 

Also, I didn't choose which project to test with so sorry if that projects cover art is a bit saucy. 

Otherwise, any other project will show the "Report this project" label as normal

### CODE REVIEW

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Library/ViewModels/ProjectPageViewModelTests.swift:767:1 | warning: (indent) Indent code in accordance with the scope level. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by ifosli — CHANGES_REQUESTED** [review]
Generally looks good, though I have some concerns. See comments. Also, a11y for this cell needs to be updated as well (mark `ReportProjectCell` as an a11y element and as a button, at least when it's not flagged) but I can fold that in to the other a11y things if you'd prefer.

File: Kickstarter-iOS/Features/ProjectPamphletContentDataSource_DEPRECATED_09_06_2022/ProjectPamphletContentDataSource.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ internal final class ProjectPamphletContentDataSource: ValueCellDataSource {
 
     let values = [
       ProjectPamphletSubpage.comments(project.stats.commentsCount as Int?, .first),
-      ProjectPamphletSubpage.updates(project.stats.updatesCount as Int?, .last)
+      ProjectPamphletSubpage.updates(project.stats.updatesCount as Int?, .middle)
```
[the remark this answers is not in the document]
Comment (scottkicks) [reply Kickstarter-iOS/Features/ProjectPamphletContentDataSource_DEPRECATED_09_06_2022/ProjectPamphletContentDataSource.swift:19]: Ah yea, good catch. was just making sure all uses of these cells had the right positioning. I'll undo this change. Looking forward to removing this file completely.

File: Kickstarter-iOS/Features/ProjectPage/Views/ReportProjectLabelView.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      HStack(alignment: .top, spacing: 10) {
+        Image("info")
+          .resizable()
+          .scaledToFit()
+          .frame(width: 20, height: 20)
+          .foregroundColor(Color(.ksr_support_500))
+
+        Text(
+          html: Strings.It_looks(
+            our_rules: HelpType.prohibitedItems
```
[the remark this answers is not in the document]
Comment (scottkicks) [reply Kickstarter-iOS/Features/ProjectPage/Views/ReportProjectLabelView.swift:41]: shoot my bad!

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/Features/ProjectPage/Views/ReportProjectLabelView.swift:7:1 | warning: (indent) Indent code in accordance with the scope level. |
| Kickstarter-iOS/Features/ProjectPage/Views/ReportProjectLabelView.swift:9:1 | warning: (indent) Indent code in accordance with the scope level. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter-iOS/Features/ProjectPage/Views/Cells/ReportProjectCell.swift [hunk 3]
```diff
@@ -1,46 +1,59 @@
...
+  // MARK: - Private Methods
+
+  private func setupTableViewCellStyle(projectFlagged: Bool) {
+    let accessibilityTraits = projectFlagged
+      ? UIAccessibilityTraits.staticText
+      : UIAccessibilityTraits.button
+
+    _ = self
+      |> baseTableViewCellStyle()
+      |> ReportProjectCell.lens.accessibilityTraits .~ accessibilityTraits
```
Comment (ifosli) [inline Kickstarter-iOS/Features/ProjectPage/Views/Cells/ReportProjectCell.swift:32]: Thanks for adding this! Unfortunately, I don't think it does anything currently; we'd need to also set `isAccessibilityElement = true` and the accessibility label. I don't see an easy way to set the label based on the swiftui child view. So I'd suggest either leaving this alone for now or adding it just for the button state. Ex:

```
if projectFlagged  {
  self.accessibilityTraits = super.accessibilityTraits
  self.isAccessibilityElement = false
} else {
  self.accessibilityTraits.insert(.button)
  self.isAccessibilityElement = true
  self.accessibilityLabel = Strings.Report_this_project_to()
}
```

**Review by ifosli — APPROVED** [review]
Size class stuff looks good - thanks for updating! Accessibility still isn't quite there, but I'll leave it up to you if you want to fix that now or not.

File: Kickstarter-iOS/Features/ProjectPage/Views/Cells/ReportProjectCell.swift (hunk 3 above)
Comment (scottkicks) [reply Kickstarter-iOS/Features/ProjectPage/Views/Cells/ReportProjectCell.swift:32]: oh, good callout! I'll look into this in the next PR

**Pull request merged**